File: app/src/main/java/com/bignerdranch/android/cartoondemo/MainActivity.java
package com.bignerdranch.android.cartoondemo;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.annotation.SuppressLint;
import android.graphics.drawable.AnimationDrawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    private Animation animation ;
    private ImageView animationImg1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        animationImg1 =  findViewById(R.id.animation1);
        animationImg1.setImageResource(R.drawable.cartoon);
//        AnimationDrawable animationDrawable1 = (AnimationDrawable) animationImg1.getDrawable();
//        animationDrawable1.start();
//
//        animation = AnimationUtils.loadAnimation(this, R.anim.scale);
//        animationImg1 .startAnimation(animation);
//
//        animation.setAnimationListener(new Animation.AnimationListener() {
//            @Override
//            public void onAnimationStart(Animation animation) {
//
//            }
//
//            @Override
//            public void onAnimationEnd(Animation animation) {
//                animationImg1 .startAnimation(animation);
//
//            }
//
//            @Override
//            public void onAnimationRepeat(Animation animation) {
//
//            }
//        });


        ObjectAnimator alphaAnim = ObjectAnimator.ofFloat( animationImg1, "alpha", 1.0f, 0.5f, 0.8f, 1.0f);
        ObjectAnimator scaleXAnim = ObjectAnimator.ofFloat( animationImg1, "scaleX", 0.0f, 1.0f);
        ObjectAnimator scaleYAnim = ObjectAnimator.ofFloat( animationImg1, "scaleY", 0.0f, 2.0f);
        ObjectAnimator rotateAnim = ObjectAnimator.ofFloat( animationImg1, "rotation", 0, 360);
        ObjectAnimator transXAnim = ObjectAnimator.ofFloat( animationImg1, "translationX", 100, 400);
        ObjectAnimator transYAnim = ObjectAnimator.ofFloat( animationImg1, "translationY", 100, 750);
        AnimatorSet set = new AnimatorSet();
//        set.playTogether(alphaAnim, scaleXAnim, scaleYAnim, rotateAnim, transXAnim, transYAnim);
                set.playSequentially(alphaAnim, scaleXAnim, scaleYAnim, rotateAnim, transXAnim, transYAnim);
        set.setDuration(3000);
        set.start();


    }
}

File: myapplication/src/main/java/com/bignerdranch/android/myapplication/MainActivity.kt
package com.bignerdranch.android.myapplication

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.graphics.drawable.AnimationDrawable
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.ImageView

class MainActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val animationImg1 = findViewById(R.id.animation1) as ImageView
        animationImg1.setImageResource(R.drawable.cartoon_kotlin)
//        val animationDrawable1 = animationImg1.drawable as AnimationDrawable
//        animationDrawable1.start()
//
//        val animation = AnimationUtils.loadAnimation(this, R.anim.scale)
//        animationImg1.startAnimation(animation)
//
//        animation.setAnimationListener(object : Animation.AnimationListener {
//            override fun onAnimationStart(animation: Animation) {
//
//            }
//
//            override fun onAnimationEnd(animation: Animation) {
//                animationImg1.startAnimation(animation)
//
//            }
//
//            override fun onAnimationRepeat(animation: Animation) {
//
//            }
//        })

        val alphaAnim = ObjectAnimator.ofFloat(animationImg1, "alpha", 1.0f, 0.5f, 0.8f, 1.0f)
        val scaleXAnim = ObjectAnimator.ofFloat(animationImg1, "scaleX", 0.0f, 1.0f)
        val scaleYAnim = ObjectAnimator.ofFloat(animationImg1, "scaleY", 0.0f, 2.0f)
        val rotateAnim = ObjectAnimator.ofFloat(animationImg1, "rotation", 0f, 360f)
        val transXAnim = ObjectAnimator.ofFloat(animationImg1, "translationX", 100f, 400f)
        val transYAnim = ObjectAnimator.ofFloat(animationImg1, "translationY", 100f, 750f)
        val set = AnimatorSet()
//        set.playTogether(alphaAnim, scaleXAnim, scaleYAnim, rotateAnim, transXAnim, transYAnim);
        set.playSequentially(alphaAnim, scaleXAnim, scaleYAnim, rotateAnim, transXAnim, transYAnim)
        set.duration = 3000
        set.start()
    }
}
